lib/src/main/java/com/diffplug/spotless/generic/IndentStep.java

```diff
@@ -49,32 +55,21 @@ public FormatterStep create(int numSpacesPerTab) {
 
 	/** Creates a step which will indent with the given type of whitespace, converting between tabs and spaces at the given ratio. */
 	public static FormatterStep create(Type type, int numSpacesPerTab) {
-		Objects.requireNonNull(type, "type");
 		return FormatterStep.create("indentWith" + type.tabSpace("Tabs", "Spaces"),
```

Comment:
My IDE tells me `type.tabSpace("Tabs", "Spaces")` can be simplified to `type.tabSpace()` if `tabSpace()` is simplified to:

```java
private String tabSpace() {
	return this == TAB ? "Tabs" : "Spaces";
}
```

WDYT?

Reply:
What I like about the current implementation is that it's clear that the final values are

- indentWithTabs
- indentWithSpaces

If we performed this refactor, it might seem like it was `indentWithTab`. It's nice when values are declared in the same place they are used, especially if they're only used once.